test/tapeimage.cpp

```diff
@@ -433,6 +426,81 @@ TEST_CASE(
     lfp_close(tif);
 }
 
+TEST_CASE(
+    "Operation past end-of-file"
+    "[tapeimage][eof]") {
+
+    const auto contents = std::vector< unsigned char > {
+        0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00,
+        0x14, 0x00, 0x00, 0x00,
+
+        /* begin body */
+        0x01, 0x02, 0x03, 0x04,
+        0x05, 0x06, 0x07, 0x08,
+        /* end body */
+
+        0x01, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00,
+        0x20, 0x00, 0x00, 0x00,
+
+        0x01, 0x00, 0x00, 0x00,
+        0x14, 0x00, 0x00, 0x00,
+        0x2C, 0x00, 0x00, 0x00,
+    };
+
+    const auto expected1 = std::vector< unsigned char > {
+        0x01, 0x02, 0x03, 0x04,
+        0x05, 0x06, 0x07, 0x08,
+        0xFF, 0xFF // never read by lfp
+    };
+
+    std::FILE* fp = std::tmpfile();
+    std::fwrite(contents.data(), 1, contents.size(), fp);
+    std::rewind(fp);
+
+    auto* cfile = lfp_cfile(fp);
+    auto* tif = lfp_tapeimage_open(cfile);
+
+    SECTION( "Read past eof" ) {
+        auto out = std::vector< unsigned char >(10, 0xFF);
+        std::int64_t bytes_read = -1;
+        const auto err = lfp_readinto(tif, out.data(), 10, &bytes_read);
+
+        CHECK(bytes_read == 8);
+        CHECK(err == LFP_EOF);
+        CHECK_THAT(out, Equals(expected1));
+
+        std::int64_t tell;
+        lfp_tell(tif, &tell);
+        CHECK(tell == 8);
+
+        lfp_close(tif);
+    }
+
+    SECTION( "Read past eof - after an seek past eof" ) {
+         auto err = lfp_seek(tif, 10);
+         CHECK(err == LFP_OK);
+
+         char x = 0xFF;
+         std::int64_t bytes_read = -1;
+         err = lfp_readinto(tif, &x, 1, &bytes_read);
+
+         CHECK(err == LFP_EOF);
+         CHECK(bytes_read == 0);
+    }
+
+    SECTION( "Seek past eof" ) {
+         const auto err = lfp_seek(tif, 10);
+         CHECK(err == LFP_OK);
+
+         std::int64_t tell = -1;
+         lfp_tell(tif, &tell);
+         CHECK(tell == 10);
+    }
```

Comment:
might be useful to do "peek and tell" to verify your claim on underlying file

> the tell of eof + the remaining requested bytes

so 34 in your case.